### ISSUE TITLE
Question about "krnl.global"()

### ISSUE THREAD

**Issue by yuucyf** [onnx/onnx-mlir#3036]
I used onnx-mlir with the --EmitMLIR parameter to generate mlir build-in dialect op from a model file. However, the generated .mlir file still contains operations like "krnl.global"(). Isn't this a krnl op rather than a built-in MLIR dialect op? Shouldn't this be considered a bug?

**Comment by AlexandreEichenberger** [comment]
No, it is not a bug, and as a matter of fact, we lower a few kernel operations late. If you wanted to have a pure mlir dialect, you would have to add a pass that adds the lowering of these krnl ops to mlir equivalent ones. 

You can search for the rules that lower these specific ops, and add a pass. We would have no issue if you wanted to upstream that code, and maybe create a new target (e.g. `-EmitOnlyMLIR` or an option like `-only-mlir` that would activate that pass when the `-EmitMLIR` target is on.

**Comment by yuucyf** [comment]
> No, it is not a bug, and as a matter of fact, we lower a few kernel operations late. If you wanted to have a pure mlir dialect, you would have to add a pass that adds the lowering of these krnl ops to mlir equivalent ones.
> 
> You can search for the rules that lower these specific ops, and add a pass. We would have no issue if you wanted to upstream that code, and maybe create a new target (e.g. `-EmitOnlyMLIR` or an option like `-only-mlir` that would activate that pass when the `-EmitMLIR` target is on.
> 
> Hope this helps. We can probably help you a bit with specific issues you may have if you start this effort and want to upstream it. I suspect other folks might be interested in that too, and we are happy to help more folks getting onboard of onnx-mlir.

However, using the -EmitMLIR option might easily lead others to misunderstand that it generates a purely MLIR dialect. Also, I’m not yet sure how many krnl operations are present in the MLIR generated by the -EmitMLIR option?

**Comment by ffly666** [comment]
This is a question, we would prefer that the `-EmitMLIR` option generate pure MLIR dialects without including krnl dialects.
